File: market-application/src/main/java/com/example/market/application/command/PlaceListingCommand.java
package com.example.market.application.command;

import com.example.market.domain.catalog.SkuId;
import com.example.market.domain.shared.Money;
import com.example.market.domain.shared.UserId;

public record PlaceListingCommand(
        String idempotencyKey,
        UserId sellerId,
        SkuId skuId,
        Money askPrice
) {}

File: market-application/src/main/kotlin/com/example/market/application/command/PlaceListingCommand.kt
package com.example.market.application.command

import com.example.market.domain.catalog.SkuId
import com.example.market.domain.shared.Money
import com.example.market.domain.shared.UserId

@JvmRecord
data class PlaceListingCommand(
    val idempotencyKey: String,
    val sellerId: UserId,
    val skuId: SkuId,
    val askPrice: Money,
)
